poly/src/evaluations/multivariate/multilinear/dense.rs

```diff
@@ -58,7 +58,7 @@ impl<F: Field> DenseMultilinearExtension<F> {
             ark_std::mem::swap(&mut a, &mut b);
         }
         assert!(
-            a + k < self.num_vars && b + k < self.num_vars,
+            a + k < self.num_vars && b + k <= self.num_vars,
```

Comment:
That looks good. A small optimisation:
if we move the `a==b` check above the first assert, then we can even skip the first part of the assert (`a + k < self.num_vars`), since at that point `b` would be strictly larger than `a`, and so `b + k <= self.num_vars` would be a sufficient condition.

Reply:
https://github.com/montekki/algebra/pull/1

Reply:
Thanks!